src/index.ts

```diff
@@ -104,7 +104,18 @@ function swc(options: PluginOptions = {}): Plugin {
         }
       ]);
 
-      return swcTransform(code, swcOption);
+      // swc cannot transform module ids with "\0" (would throw
+      // "error: Unterminated string constant"), which can be used by
+      // some plugins (e.g. "@rollup/plugin-commonjs"), see:
+      // - https://rollupjs.org/guide/en/#conventions
+      // - https://github.com/rollup/plugins/blob/02fb349d315f0ffc55970fba5de20e23f8ead881/packages/commonjs/src/helpers.js#L15
+      return swcTransform(code.replace('\0', '\\0'), swcOption)
```

Comment:
```suggestion
      return swcTransform(code.replaceAll('\0', '\\0'), swcOption)
```

Use `String.prototype.replaceAll` or `/\0/g`. There might be more than one virtual module being imported.

Also, `\\0` is kinda dangerous: maybe someone would use `\\0` in their code, and the plugin might unexpectedly unescape it. I'd prefer something like `$$__SECRET_SLASH_0_ESCAPE_IDENTIFIER_DO_NOT_USE_OR_YOU_WILL_BE_FIRED__$$`

Reply:
You are right. Did not spend further thoughts on that change after you opened the bug report at swc. Will adjust it as suggested.

Reply:
Applied the changes here: https://github.com/SukkaW/rollup-plugin-swc/pull/2/commits/9bf7b4e170c6240e109313b929f2f0ca192e0bc2